### PR TITLE
Backend route to run code

### PR DESCRIPTION
**Pull request by nuddin175**
- Added route /runCode to execute code (updated to post route)
- Currently supports python and javascript

image
image
image

- Integrated with the front end to display code output
image

### CODE REVIEW

File: client/src/Components/OutputConsole.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React, { useState } from 'react';
+import { RunCodeButton } from './Buttons';
+import { useStyles } from '../themes/theme';
+import axios from 'axios';
+
+const OutputConsole = (props) => {
+  const classes = useStyles();
+  const [output, setOutput] = useState('');
+
+  // const runCode = async () => {
```
Comment (dahrk) [inline client/src/Components/OutputConsole.js:10]: can remove for push

Comment (skassam21) [reply client/src/Components/OutputConsole.js:10]: agreed, we should remove this

File: client/src/pages/Interview.js [hunk 2]
```diff
@@ -1,47 +1,56 @@
...
+import InterviewQuestionDetails from '../components/InterviewQuestionDetails';
+import TextEditor from '../components/TextEditor';
+import InterviewHeader from '../components/InterviewHeader';
+import OutputConsole from '../components/OutputConsole';
+import axios from 'axios';
 
 const Interview = () => {
   const classes = useStyles();
+  const [codeSnippet, setCodeSnippet] = useState('');
+  //const [language, setLanguage] = useState('javascript');
```
Comment (skassam21) [inline client/src/pages/Interview.js:13]: remove this unneeded commented out code

File: client/src/pages/Interview.js [hunk 3]
```diff
@@ -1,47 +1,56 @@
...
+    setCodeData({ ...codeData, code: codeSnippet });
+  };
+
+  const handleLanguageChange = (language) => {
+    setCodeData({ ...codeData, language: language });
+  };
+
+  const runCode = async () => {
+    try {
+      const result = await axios.post(`http://localhost:3001/runCode`, codeData);
```
Comment (skassam21) [inline client/src/pages/Interview.js:32]: API calls should be made without localhost:3001

File: server/app.js [hunk 4]
```diff
@@ -37,6 +42,7 @@ app.use('/ping', pingRouter);
 app.use('/users', users);
 app.use('/interviews', interviews);
 app.use('/questions', questions);
+app.use('/runCode', executeCode);
```
Comment (skassam21) [inline server/app.js:45]: This should be using the middleware and should be mentioned in the routes file

File: server/routes/executeCode.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+const axios = require('axios');
+const router = express.Router();
+
+const apiMethod = (req, res) => {
+  let code = req.body.code;
+  let language = req.body.language;
+
+  //console.log(req);
+
+  console.log(`${language} ---- ${code}`);
```
Comment (skassam21) [inline server/routes/executeCode.js:11]: unneeded console log at the moment

File: server/routes/executeCode.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
+const express = require('express');
+const axios = require('axios');
+const router = express.Router();
+
+const apiMethod = (req, res) => {
```
Comment (skassam21) [inline server/routes/executeCode.js:5]: This logic should be written in the controllers

Comment (nuddin175) [reply server/routes/executeCode.js]: Thanks! I removed all the console logs, and updated the route to use passport and moved the function to a controller.

**Review by skassam21 — APPROVED** [review]

**Pull request merged**